From 3025c90835496475d361b4ffee995836d175c2e2 Mon Sep 17 00:00:00 2001
From: Igor Grinberg <igrinberg@habana.ai>
Date: Tue, 08 Jan 2019 14:39:15 +0200
Subject: [PATCH] [SW-0] pll: use variable for offset calculations

Switch to use variable for offset calculations instead of preforming
the same calculation multiple times.

Change-Id: I68c9bcba700e2da320533f3bb0bcea00761701f3
Signed-off-by: Igor Grinberg <igrinberg@habana.ai>
---

diff --git a/pll.c b/pll.c
index 28bad84..c57ac08 100644
--- a/pll.c
+++ b/pll.c
@@ -134,18 +134,19 @@
 
 static void hl_pll_bypass(struct hl_pll_cfg *cfg, bool bypass)
 {
-	int sel, offset;
+	int sel;
+	void *reg;
 
 	for (sel = 0; sel < HL_PLL_DIV_SIZE; sel++) {
-		offset = sel * sizeof(u32);
-
 		if (cfg->div_factor[sel] == -1)
 			continue;
 
+		reg = cfg->base + HL_PLL_DIV_SEL + (sel * sizeof(u32));
+
 		if (bypass)
-			writel(0, cfg->base + offset + HL_PLL_DIV_SEL);
+			writel(0, reg);
 		else
-			writel((sel || (cfg->div_factor[sel]))?3:1, cfg->base + offset + HL_PLL_DIV_SEL);
+			writel((sel || (cfg->div_factor[sel]))?3:1, reg);
 	}
 
 	/* just to be on a safe side, wait for bypass to propagate */
@@ -154,18 +155,18 @@
 
 static void hl_pll_div_init(struct hl_pll_cfg *cfg)
 {
-	int sel, offset;
+	int sel;
+	void *div_base;
 
 	for (sel = 0; sel < HL_PLL_DIV_SIZE; sel++) {
-		offset = sel * sizeof(u32);
-
 		if (cfg->div_factor[sel] == -1)
 			continue;
 
-		writel(cfg->div_factor[sel], cfg->base +
-						offset + HL_PLL_DIV_FACTOR);
-		writel(1, cfg->base + offset + HL_PLL_DIV_FACTOR_CMD);
-		writel(1, cfg->base + offset + HL_PLL_DIV_EN);
+		div_base = cfg->base + (sel * sizeof(u32));
+
+		writel(cfg->div_factor[sel], div_base + HL_PLL_DIV_FACTOR);
+		writel(1, div_base + HL_PLL_DIV_FACTOR_CMD);
+		writel(1, div_base + HL_PLL_DIV_EN);
 	}
 
 	/* just to be on a safe side, wait for DIV init to propagate */
